integration-tests/bats/init.bats

```diff
@@ -152,6 +152,55 @@ teardown() {
   [ "$status" -eq 1 ]
 }
 
+@test "init: running init with the new format, creates a new format database" {
+    set_dolt_user "baz", "baz@bash.com"
+
+    run dolt init --new-format
+    [ $status -eq 0 ]
+
+    run dolt init
+    [ "$status" -eq 1 ]
+
+    run cut -d ":" -f 2 .dolt/noms/manifest
+    [ "$output" = "__DOLT__" ]
+}
+
+@test "init: initing a new database displays the correct version" {
+    set_dolt_user "baz", "baz@bash.com"
+
+    run dolt init --new-format
+    [ $status -eq 0 ]
+
+    run dolt version
+    [ $status -eq 0 ]
+    [[ $output =~ "database storage format: NEW ( __DOLT__ )" ]] || false
+
+    run dolt sql -q "SELECT dolt_storage_format();"
+    [[ $output =~ "NEW ( __DOLT__ )" ]] || false
+}
+
+@test "init: initing an old database displays the correct version" {
+    set_dolt_user "baz", "bazbash.com"
+
+    run dolt init
+    [ "$status" -eq 0 ]
+
+    run dolt version
+    [ "$status" -eq 0 ]
+    [[ $output =~ "database storage format: OLD ( __LD_1__ )" ]] || false
+
+    run dolt sql -q "SELECT dolt_storage_format();"
+    [[ $output =~ "OLD ( __LD_1__ )" ]] || false
+}
+
+@test "init: empty database folder displays no version" {
```

Comment:
It's worth thinking about any testing for multi-repo environments at both the command line and through SQL, especially since we so frequently hit problems with multi-db envs. I know we have some checks in place now to ignore different formats, but I'm guessing there are still some sharp edges in there and some more testing might help root some of those out.